tt_eager/tt_dnn/op_library/ccl/reduce_scatter/host/reduce_scatter_full_worker_grid.cpp

```diff
@@ -350,6 +351,12 @@ static void add_worker_config_to_edm_builders(
                 device->worker_core_from_logical_core(worker_cores.at(w)).y));
         }
 
+        // Get the expected message size in bytes for this worker
+        auto worker_slice_shape = tensor_slicer.get_worker_slice_shapes().at(global_worker_idx);
+        uint32_t expected_message_size_bytes = worker_slice_shape.x *
```

Comment:
I would like the message size calculation to be abstracted behind another interface. Either the tensor slicer or into a Shape object (this is probably ideal). This way, when we add 4D tensor slices, which are needed for generality, we won't have to track down a bug that will be exposed on this line when that transition is made.

Since it's on the TODO list to move to a Shape in the future, I'm thinking it's fine for now to just add a method to the tensor slicer which we can deprecate when we migrate to Shape for the members of the slicer.